File: 0014-longest-common-prefix/0014-longest-common-prefix.ts
function longestCommonPrefix(strs: string[]): string {
    // find the longest word
    let prefix = strs.reduce((acc, str) => str.length > acc.length ? str : acc);

    for(let str of strs) {
        while(str.slice(0, prefix.length) != prefix) {
            prefix = prefix.slice(0, -1);
        }
    }
    return prefix;
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.rs
impl Solution {
    pub fn longest_common_prefix(strs: Vec<String>) -> String {
    // get the length of the shortest string in the array
      let mut n = strs.iter().map(|s| s.len()).min().unwrap();
    // keep reducing the length by comparing any two strings in the strs array
    // and checking for the common prefix, we reduce n until we find it
    
        while strs.windows(2).any(|s| s[0][..n] != s[1][..n]) { n -= 1}
        //return the longest common prefix
        strs[0][..n].into()
    }
}